### PR TITLE
Track losses with tensorboard

### PR DESCRIPTION
**Pull request by SwamiKannan**
# Description

This PR allows us to track progress of the loss for each step and for each epoch with Tensorboard. It is specifically written to minimize changes to the existing repo. Hence, if the SentenceTransformer repo finally updates with a reporting framework, these changes can be reverted with just two lines and the deletion of one file.

Added screenshot of tensorboard report running with this script

## Type of Change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Added new notebook (that tests end-to-end)
- [X] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

Added screenshot of my test run

tensorboard

### CODE REVIEW

File: llama-index-finetuning/llama_index/finetuning/embeddings/sentence_transformer.py [hunk 1]
```diff
@@ -1,91 +1,97 @@
...
-
-from llama_index.core.base.embeddings.base import BaseEmbedding
-from llama_index.core.embeddings.utils import resolve_embed_model
-from llama_index.finetuning.embeddings.common import (
-    EmbeddingQAFinetuneDataset,
-)
-from llama_index.finetuning.types import BaseEmbeddingFinetuneEngine
-
-
-class SentenceTransformersFinetuneEngine(BaseEmbeddingFinetuneEngine):
```
Comment (hatianzhang) [inline llama-index-finetuning/llama_index/finetuning/embeddings/sentence_transformer.py:13]: what is the change here for this file?

Comment (SwamiKannan) [reply llama-index-finetuning/llama_index/finetuning/embeddings/sentence_transformer.py:13]: So if a log_path is provided, 
  self.model will be a **TB**SentenceTransformer object
else if no log_path is provided,
  self.model will be a **standard** SentenceTransformer object

Lines 40-43 in sentence_transformer.py

```
if log_path:
            self.model = TBSentenceTransformer(model_id, writer_path = log_path)
        else:
            self.model = SentenceTransformer(model_id)
```

File: sentence_transformer.py
Comment (logan-markewich) [inline sentence_transformer.py:1]: did you mean to commit these two files in the root?

Comment (SwamiKannan) [reply sentence_transformer.py:1]: Yes, there were two files to be committed. 
These were to be in the same folder as sentence_transformer.py i.e. in finetuning\embeddings\

Comment (SwamiKannan) [reply sentence_transformer.py:1]: Do correct me if I have missed something?

Comment (logan-markewich) [reply sentence_transformer.py:1]: I don't think these files belong in the root of the entire repo 👀

Comment (SwamiKannan) [reply sentence_transformer.py:1]: oh god.... i'm so sorry... that was a mistake. Resolved on my repo so I am assuming it is replicated in this PR? 
Apologies. This is the first time I am committing to such a large and useful repo so I may be a little lost

**Review by SwamiKannan — COMMENTED** [review]
Deleted an extraneous "from" syntax

**Pull request closed without merging**